contracts/yield-source/YearnV2YieldSource.sol

```diff
@@ -60,40 +67,66 @@ contract YearnV2YieldSource is IYieldSource, ERC20Upgradeable, OwnableUpgradeabl
         uint256 amount
     );
 
-    /// @notice Initializes the yield source with 
-    /// @param _vault YearnV2 Vault in which the Yield Source will deposit `token` to generate Yield
-    /// @param _token Underlying Token / Deposit Token
+    /// @notice Mock Initializer to initialize implementations used by minimal proxies.
+    function freeze() public initializer {
+        //no-op
+    }
+
+    /// @notice Initializes the yield source with
+    /// @dev `isContract()` will return false if an attacker is calling from a constructor https://ethereum.stackexchange.com/a/64340
+    /// @param _vault Yearn V2 Vault in which the Yield Source will deposit `token` to generate Yield
+    /// @param _decimals Number of decimals the shares (inherited ERC20) will have.  Same as underlying asset to ensure same ExchangeRates.
+    /// @param _symbol Token symbol for the underlying ERC20 shares (eg: yvysDAI).
+    /// @param _name Token name for the underlying ERC20 shares (eg: PoolTogether Yearn V2 Vault DAI Yield Source).
+    /// @param _owner Yearn V2 Vault Yield Source owner.
     function initialize(
         IYVaultV2 _vault,
-        IERC20Upgradeable _token
-    ) 
-        public 
+        uint8 _decimals,
+        string calldata _symbol,
+        string calldata _name,
+        address _owner
+    )
+        public
         initializer
+        returns (bool)
     {
-        require(address(vault) == address(0), "YearnV2YieldSource:: already initialized");
-        require(_vault.token() == address(_token), "YearnV2YieldSource:: incorrect vault");
-        require(_vault.activation() != uint256(0), "YearnV2YieldSource:: vault not initialized");
+        require(address(_vault).isContract(), "YearnV2YieldSource/vault-not-contract-address");
+        require(_vault.activation() != uint256(0), "YearnV2YieldSource/vault-not-initialized");
+
         // NOTE: Vaults from 0.3.2 to 0.3.4 have dips in shareValue
-        require(!areEqualStrings(_vault.apiVersion(), "0.3.2"), "YearnV2YieldSource:: vault not compatible");
-        require(!areEqualStrings(_vault.apiVersion(), "0.3.3"), "YearnV2YieldSource:: vault not compatible");
-        require(!areEqualStrings(_vault.apiVersion(), "0.3.4"), "YearnV2YieldSource:: vault not compatible");
+        string memory _vaultAPIVersion = _vault.apiVersion();
+
+        require(!_areEqualStrings(_vaultAPIVersion, "0.3.2"), "YearnV2YieldSource/vault-not-compatible");
+        require(!_areEqualStrings(_vaultAPIVersion, "0.3.3"), "YearnV2YieldSource/vault-not-compatible");
+        require(!_areEqualStrings(_vaultAPIVersion, "0.3.4"), "YearnV2YieldSource/vault-not-compatible");
 
         vault = _vault;
-        token = _token;
+        token = IERC20Upgradeable(_vault.token());
 
         __Ownable_init();
+        transferOwnership(_owner);
```

Comment:
@PierrickGT  Similar to other contracts in the PT family the `transferOwnership` function is called outside of the initializer function. Perhaps we want to follow that pattern here?

For example this might be done so a builder (or alternative deploy process) can still call other onlyOwner protected functions required for setup before transferring ownership.

In this case calling `setMaxLosses` during the setup process.

So something like this.

```
contract.initialize(....)
contract.setMaxLosses(....)
contract.transferOwnership(....)
```